### PR TITLE
Fix MouseEvent `.buttons` not including triggering mouse button

### LINKED ISSUE #2344: Mouse event behaviour
```
    def on_mouse_press(self, event):    
        print(event.buttons)
```

The [docs](https://vispy.org/api/vispy.app.canvas.html?highlight=mouseevent#vispy.app.canvas.MouseEvent) says 

```
buttons[int, …]
The list of buttons pressed during this event.
```

But, pressing the left click returns an empty list and while holding down the left click and pressing other mouse buttons returns the mouse button key [1], [2] etc.

Is it a bug or I am getting something wrong?

I am using `glfw` as my backend.

### PR DESCRIPTION
**Pull request by tushar5526**
Closes #2344 
- Explicitly add button to the buttons list in mouse event

Doc Changes
- Fixed a typo

### CODE REVIEW

File: vispy/app/backends/_glfw.py [hunk 1]
```diff
@@ -27,7 +27,7 @@
 try:
     import glfw
 except ImportError:
-    why_not = "Could not import glwf, you may need to `pip install glfw` first."
+    why_not = "Could not import glfw, you may need to `pip install glfw` first."
```
Comment (djhoese) [inline vispy/app/backends/_glfw.py:30]: :man_facepalming: It only took 8 years or so to find this typo. Nice catch and thank you.

File: vispy/app/canvas.py [hunk 2]
```diff
@@ -632,7 +632,9 @@ def __init__(self, type, pos=None, button=None, buttons=None,
         Event.__init__(self, type, **kwargs)
         self._pos = np.array([0, 0]) if (pos is None) else np.array(pos)
         self._button = int(button) if (button is not None) else None
-        self._buttons = [] if (buttons is None) else buttons
+        # Explicitly add button to buttons
+        self._buttons = [] if (buttons is None) else buttons + (
+            [button] if button is not None and type == 'mouse_press' else [])
```
Comment (djhoese) [inline vispy/app/canvas.py:637]: I found this bit a little hard to understand. Does splitting it into multiple lines and updating the comment still make sense?

```suggestion
        # Explicitly add button to buttons if newly pressed
        newly_pressed_buttons = [button] if button is not None and type == 'mouse_press' else []
        self._buttons = [] if (buttons is None) else buttons + newly_pressed_buttons
```

**Comment by djhoese** [comment]
Note: The CI will likely fail due to some issues we're having with the latest github ubuntu image.

File: vispy/app/backends/_glfw.py (hunk 1 above)
Comment (tushar5526) [reply vispy/app/backends/_glfw.py:30]: Haha, I forgot to install glfw and "glwf" popped up :p

File: vispy/app/canvas.py (hunk 2 above)
Comment (tushar5526) [reply vispy/app/canvas.py:637]: Would it be fine, if I add the button in a new line. Something like 

```
        self._buttons += [button] if button is not None and type == 'mouse_press' else []

```

Comment (tushar5526) [reply vispy/app/canvas.py:637]: I will update the comment, the best would be to mention this issue if its allowed.

Comment (djhoese) [reply vispy/app/canvas.py:637]: The only problem with doing that `+=` is that it modifies the original `buttons` list inplace, right? We don't want to modify the callers arguments if possible. You could do `self._buttons = self._buttons + ...`. The only argument I have for my way is that the variable name gives some context for what is being done and why (possibly making the comment unnecessary).

If you want to add a comment linking to the issue then that's fine by me.

Comment (tushar5526) [reply vispy/app/canvas.py:637]: Yes, it would modify the original arguments. Thanks it completely skipped my mind. 

I will make the changes.

**Comment by tushar5526** [comment]
I have rebased to main and where should I put the `mouse_event` test, under the utils or app tests?

**Comment by djhoese** [comment]
I would say in the app tests. Either here: https://github.com/vispy/vispy/blob/main/vispy/app/tests/test_canvas.py or in a new `test_events.py` module...eh maybe `test_canvas.py` is best for now (but a new class or new function in there).

**Comment by tushar5526** [comment]
I haven't pushed the rebased branch yet, will do it soon along with the tests.

**Comment by tushar5526** [comment]
@djhoese I have added very basic tests for MouseEvent, I couldn't find a way to mimic mouse clicks without external libraries, if there is something I could call to mimic a click on my window then I could update the tests.

**Comment by djhoese** [comment]
I've never done it inside a vispy test (I've done it on one of my own Qt applications). I think this will just have to be "good enough". I'm sure we could call some private (`_` prefixed) methods in the backend and make it work, but I'm not sure it is worth it at that point. Let's see how the tests go and then get one last review from a few people.

@brisvag @almarklein any opinions?

**Review by djhoese — APPROVED** [review]

**Review by brisvag — APPROVED** [review]
Seems all good to me!

**Comment by djhoese** [comment]
Thanks @tushar5526!

**Pull request merged**